src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -206,22 +222,14 @@ impl Default for WorkunitMetadata {
 }
 
 /// Abstract id for passing user metadata items around
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[derive(Clone, Debug)]
 pub enum UserMetadataItem {
-  PyValue(UserMetadataPyValue),
+  // NB: Always stores a ...
```

Comment:
TODO?

Reply:
Oops, yep.

As to your other comment: the traits try to explain themselves:
> // NB: Only implemented for `Value`, but is boxed to avoid a cycle between this crate and the
> // `engine` crate.